### PR TITLE
Consolidate Ansibleee controller logic

### PR DESCRIPTION
**Pull request by bshephar**
This change consolidates the openstack-ansibleee-operator logic into the openstack-operator.
The openstack-ansibleee-operator currently is a full operator just to create Kubernetes jobs. It would be better if this functionality was just included with the rest of the dataplane code. This change keeps the controller, but runs it under the openstack-operator instead of standalone.

Jira: https://issues.redhat.com/browse/OSPRH-8718

### CODE REVIEW

**Comment by bshephar** [comment]
recheck

**Comment by bshephar** [comment]
recheck

**Review by fao89 — COMMENTED** [review]
we need to add the API here:
https://github.com/openstack-k8s-operators/openstack-operator/blob/main/PROJECT
and add the webhook here:
https://github.com/openstack-k8s-operators/openstack-operator/blob/main/hack/configure_local_webhook.sh#L31
https://github.com/openstack-k8s-operators/openstack-operator/blob/main/hack/clean_local_webhook.sh

File: main.go
Comment (fao89) [inline main.go:1]: we need to initialize the reconciler: https://github.com/openstack-k8s-operators/openstack-ansibleee-operator/blob/main/main.go#L123-L130

Comment (bshephar) [reply main.go:1]: Could have sworn I did this. Maybe it was just the suite_test.go file that I did it in and left this one out. Thanks!

File: apis/bases/dataplane.openstack.org_openstackansibleees.yaml [hunk 1]
```diff
@@ -0,0 +1,1596 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.11.1
+  creationTimestamp: null
+  name: openstackansibleees.dataplane.openstack.org
+spec:
+  group: dataplane.openstack.org
```
[the remark this answers is not in the document]
Comment (bshephar) [reply apis/bases/dataplane.openstack.org_openstackansibleees.yaml:10]: Initially, I was kind of thinking along the lines of just fully consolidating it with the dataplane. But it is probably easier to not mess with that and just keep it as its own API group. I've amended the PR now to ensure the group doesn't change.

**Comment by bshephar** [comment]
Still got an RBAC issue here:
```
2024-07-20T04:03:49.689Z	ERROR	Status update failed	{"controller": "openstackansibleee", "controllerGroup": "ansibleee.openstack.org", "controllerKind": "OpenStackAnsibleEE", "OpenStackAnsibleEE": {"name":"repo-setup-edpm-deployment-openstack-edpm-ipam","namespace":"openstack"}, "namespace": "openstack", "name": "repo-setup-edpm-deployment-openstack-edpm-ipam", "reconcileID": "8519a76a-2bb3-4c62-aa1f-d0e234dfe52e", "error": "openstackansibleees.ansibleee.openstack.org \"repo-setup-edpm-deployment-openstack-edpm-ipam\" is forbidden: User \"system:serviceaccount:openstack-operators:openstack-operator-controller-manager\" cannot patch resource \"openstackansibleees/status\" in API group \"ansibleee.openstack.org\" in the namespace \"openstack\""}
github.com/openstack-k8s-operators/lib-common/modules/common/helper.(*Helper).PatchInstance
	/opt/app-root/src/go/pkg/mod/github.com/openstack-k8s-operators/lib-common/modules/common@v0.4.1-0.20240715111029-358ba8041494/helper/helper.go:225
github.com/openstack-k8s-operators/openstack-operator/controllers/ansibleee.(*OpenStackAnsibleEEReconciler).Reconcile.func1
	/remote-source/controllers/ansibleee/openstack_ansibleee_controller.go:124
github.com/openstack-k8s-operators/openstack-operator/controllers/ansibleee.(*OpenStackAnsibleEEReconciler).Reconcile
	/remote-source/controllers/ansibleee/openstack_ansibleee_controller.go:167
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.6/pkg/internal/controller/controller.go:119
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.6/pkg/internal/controller/controller.go:316
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.6/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.6/pkg/internal/controller/controller.go:227

```

**Comment by bshephar** [comment]
recheck

**Comment by bshephar** [comment]
recheck

**Comment by bshephar** [comment]
Unrelated failure in redhat-operators marketplace

**Comment by bshephar** [comment]
```
  stderr: |-
    timed out waiting for the condition on pods/certified-operators-j72vz
    timed out waiting for the condition on pods/certified-operators-s9m9b
    timed out waiting for the condition on pods/community-operators-nm9f9
    timed out waiting for the condition on pods/community-operators-qsdxj
    timed out waiting for the condition on pods/marketplace-operator-8b455464d-2wg58
    timed out waiting for the condition on pods/redhat-marketplace-fnk59
    timed out waiting for the condition on pods/redhat-marketplace-g9bwq
    timed out waiting for the condition on pods/redhat-operators-bnp45
    timed out waiting for the condition on pods/redhat-operators-plcgt
```

**Comment by bshephar** [comment]
I think I'll close this one and we'll just completely remove the AEE CR. We can just replace the entire thing by directly creating jobs from the Dataplane controllers

**Comment by fao89** [comment]
+1 to create jobs from dataplane controllers

File: config/crd/bases/dataplane.openstack.org_openstackansibleees.yaml
Comment (fao89) [inline config/crd/bases/dataplane.openstack.org_openstackansibleees.yaml:1]: we should remove this file

File: apis/bases/dataplane.openstack.org_openstackansibleees.yaml
Comment (fao89) [inline apis/bases/dataplane.openstack.org_openstackansibleees.yaml:1]: we should remove this file

File: main.go
Comment (fao89) [inline main.go:1]:
```
2024-07-22T06:25:16.656Z	ERROR	Reconciler error	{"controller": "openstackansibleee", "controllerGroup": "ansibleee.openstack.org", "controllerKind": "OpenStackAnsibleEE", "OpenStackAnsibleEE": {"name":"repo-setup-edpm-deployment-ipam-openstack-edpm-ipam","namespace":"openstack"}, "namespace": "openstack", "name": "repo-setup-edpm-deployment-ipam-openstack-edpm-ipam", "reconcileID": "df44b526-af73-462f-8521-b958100399d5", "error": "Error getting network-attachment-definition ctlplane/openstack - no kind is registered for the type v1.NetworkAttachmentDefinition in scheme \"go/pkg/mod/k8s.io/apimachinery@v0.28.11/pkg/runtime/scheme.go:100\""}
```

it seems you need to add the NetworkAttachmentDefinition scheme, or maybe reorder and have glance or any other service operator that initializes it, to be before AEE scheme https://github.com/openstack-k8s-operators/glance-operator/blob/main/main.go#L65

**Pull request closed without merging**